File: LC/Python/Medium/2294.py
class Solution:
    def partitionArray(self, nums: List[int], k: int) -> int:
        nums.sort()
        ans = 1

        curr_min = nums[0]
        for num in nums[1:]:
            if num - curr_min > k:
                ans += 1
                curr_min = num

        return ans

File: LC/C++/Medium/2294.cpp
class Solution {
public:
    int partitionArray(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        int ans = 1;
        int curr_min = nums[0];

        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] - curr_min > k) {
                ans += 1;
                curr_min = nums[i];
            }
        }

        return ans;
    }
};